projects/plugins/jetpack/modules/publicize/publicize.php

```diff
@@ -713,7 +829,7 @@ public function get_filtered_connection_data( $selected_post_id = null ) {
 				 * If this is a global connection and this user doesn't have enough permissions to modify
 				 * those connections, don't let them change it.
 				 */
-				if ( ! $done && ( 0 == $connection_data['user_id'] && ! current_user_can( $this->GLOBAL_CAP ) ) ) {
+				if ( ! $done && ( 0 === (int) $connection_data['user_id'] && ! current_user_can( $this->GLOBAL_CAP ) ) ) {
```

Comment:
! is_global_connection( ) check my be instead of 0 === (int) $connection_data['user_id']?